File: examples/go/shift-ops/ShiftOps_test.go
package contract

import (
	"testing"

	runar "github.com/icellan/runar/packages/runar-go"
)

func TestShiftOps_TestShift(t *testing.T) {
	c := &ShiftOps{A: 16}
	c.TestShift() // should not panic
}

func TestShiftOps_Compile(t *testing.T) {
	if err := runar.CompileCheck("ShiftOps.runar.go"); err != nil {
		t.Fatalf("Rúnar compile check failed: %v", err)
	}
}

File: examples/rust/shift-ops/ShiftOps_test.rs
#[path = "ShiftOps.runar.rs"]
mod contract;

use contract::*;

#[test]
fn test_shift() {
    let c = ShiftOps { a: 16 };
    c.test_shift();
}

#[test]
fn test_compile() {
    runar::compile_check(
        include_str!("ShiftOps.runar.rs"),
        "ShiftOps.runar.rs",
    )
    .unwrap();
}
